packages/astro/src/vite-style-transform/style-transform.ts

```diff
@@ -35,13 +39,60 @@ export function createTransformStyles(
 	const normalizedID = getNormalizedIDForPostCSS(filename);
 
 	return async function (styleSource, lang) {
-		const result = await viteStyleTransformer.transformStyleWithVite.call(pluginContext, {
-			id: normalizedID,
-			source: styleSource,
-			lang,
-			ssr,
-			viteDevServer: viteStyleTransformer.viteDevServer,
-		});
+		let result: any;
+		try {
+			result = await viteStyleTransformer.transformStyleWithVite.call(pluginContext, {
+				id: normalizedID,
+				source: styleSource,
+				lang,
+				ssr,
+				viteDevServer: viteStyleTransformer.viteDevServer,
+			});
+		} catch (err: any) {
+			const fileContent = readFileSync(filename).toString();
+			const styleTagBeginning = fileContent.indexOf(err.input?.source ?? err.code);
+
+			if (err.name === 'CssSyntaxError') {
+				const errorLine = positionAt(styleTagBeginning, fileContent).line + (err.line ?? 0);
+				const cssError = new CSSError({
+					errorCode: AstroErrorCodes.CssSyntaxError,
+					message: err.reason,
+					location: {
+						file: filename,
+						line: errorLine,
+						column: err.column,
+					},
+				});
+
+				// Vite will handle creating the frame for us with proper line numbers, no need to create one
```

Comment:
This took me by surprise while writing this PR, it turns out that Vite is able to create codeframes by itself whenever an error happens in parts it understands

Sometimes they're better than ours, sometimes not, luckily it checks if we already have one before doing it, so we can case by case it